app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerFragment.kt

```diff
@@ -210,148 +130,210 @@ class MenuDrawerFragment : MenuFoldersFragment(), MailboxListFragment {
         )
     }
 
-    override fun setupAdapters() {
-        super.setupAdapters()
-        binding.mailboxList.adapter = MailboxesAdapter(
-            isInMenuDrawer = isInMenuDrawer,
-            hasValidMailboxes = hasValidMailboxes,
-            onValidMailboxClicked = { mailboxId -> onValidMailboxClicked(mailboxId) },
-            onLockedMailboxClicked = { mailboxEmail -> onLockedMailboxClicked(mailboxEmail) },
-            onInvalidPasswordMailboxClicked = { mailbox -> onInvalidPasswordMailboxClicked(mailbox) },
+    private fun setupRecyclerView() {
+        binding.menuDrawerRecyclerView.adapter = menuDrawerAdapter(
+            currentClassName = currentClassName,
+            onAskingTransition = ::executeMenuDrawerTransition,
+            onAskingToCloseDrawer = ::closeDrawer,
+            onMailboxesHeaderClicked = ::onMailboxesHeaderClicked,
+            onValidMailboxClicked = ::onValidMailboxClicked,
+            onInvalidPasswordMailboxClicked = ::onInvalidPasswordMailboxClicked,
+            onLockedMailboxClicked = ::onLockedMailboxClicked,
+            onCustomFoldersHeaderClicked = ::onCustomFoldersHeaderClicked,
+            onCreateFolderClicked = ::onCreateFolderClicked,
+            onFolderClicked = ::onFolderSelected,
+            onCollapseChildrenClicked = ::onFolderCollapsed,
+            onSyncAutoConfigClicked = ::onSyncAutoConfigClicked,
+            onImportMailsClicked = ::onImportMailsClicked,
+            onRestoreMailsClicked = ::onRestoreMailsClicked,
+            onFeedbackClicked = ::onFeedbackClicked,
+            onHelpClicked = ::onHelpClicked,
+            onAppVersionClicked = ::onAppVersionClicked,
         )
     }
 
-    override fun onFolderSelected(folderId: String) {
-        mainViewModel.openFolder(folderId)
-        closeDrawer()
-    }
-
-    override fun onFolderCollapse(folderId: String, shouldCollapse: Boolean) {
-        menuDrawerViewModel.toggleFolderCollapsingState(folderId, shouldCollapse)
-    }
-
-    private fun addMenuDrawerTransition(shouldExclude: Boolean = false, shouldFade: Boolean = true) {
+    private fun executeMenuDrawerTransition() {
         val transition = TransitionSet()
             .addTransition(ChangeBounds())
-            .also { if (shouldFade) it.addTransition(Fade(IN)) }
-            .excludeTarget(RecyclerView::class.java, shouldExclude)
+            .addTransition(Fade(IN))
             .setDuration(MENU_DRAWER_TRANSITION_DURATION)
 
-        TransitionManager.beginDelayedTransition(binding.drawerContentScrollView, transition)
+        TransitionManager.beginDelayedTransition(binding.menuDrawerRecyclerView, transition)
```

Comment:
Shouldn't this temporary animation fix from @FabianDevel be gone altogether now that we have a RecyclerView that handles the items beautifully?

Reply:
This was confirmed with him directly,  it's indeed not needed anymore, this should be removed